### PR TITLE
V2.4 Pins Mod

### PR DESCRIPTION
**Pull request by hartk1213**
* [x] The mod, firmware configuration or slicer profile is in the correct category
    folder. Printable mods go to `printer_mods/`, firmware configurations
    go to `firmware_configurations/`, slicer profiles go to `slicer_profiles/`.
    Create a subfolder with your name, and place the mods in a subfolder with
    a descriptive name within that folder, e.g.: `/printer_mods/FHeilmann/flux_capacitor`
  * [x] Folders names MUST NOT contain spaces. If possible, make sure file names also 
    do not contain any spaces.
  * [x] For each mod, add a small `README.md` file to its folder with a short description
    of what the mod accomplishes. This readme can be used to add pictures, give assembly
    instructions or specify a bill of materials if the mod requires additional hardware.
  * [x] The PR modifies the top-level `README.md` of the category folder adding the 
    contribution to the table. Read the top part of the file for instructions on how
    to do this. Please preserve the alphabetical ordering while adding new rows. Make sure
    to fill out the compatibility matrix to indicate which versions of the Voron printer
    the submission is compatible with.
  * [x] The mod/configuration/profile has been tested by the person submitting the mod 
    and/or other Voron users. Make sure to add information about how the mod was tested below. 
  * [x] The mod is not merely a slight modification of an official Voron part, configuration 
    or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile 
    which only modifies a few values). *(When in doubt, contact one of the admins in the 
    Voron discord before submitting the PR)*
  * [x] Submitted STLs are printable without support. *(If the mod does not meet this criterion
    join the Voron discord and ask the other users for advice on how to modify the mod such 
    that it does not require supports)*
  * [x] Submitted STL files are not corrupt. *(This can be tested by opening the STL in PrusaSlicer
    and checking if mesh errors are reported.)*
  * [x] Submitted STL files are oriented and scaled properly for printing.
  * [x] Submitted firmware configs or slicer profiles contain no sensitive data (e.g. API keys).

#### Which mods/configurations/profiles are added by this PR?
This mod is for the V2, it involves printing new AB drive frames, new XY Joints, new Front Tensioners, and new Z idlers

The pins have replaced m5 screws, at any location where the screw acts like a shaft, holding bearings, pulleys or idlers. This allows for smoother rotation and stops the possibility of motion parts being caught in threads. This work was done by everyone, thanks to ABS plastic and voron printers for the fast iteration, @doomweasel? they fall out yet?

#### How was it tested? 
Tested by multiple users over the course of a couple months

### CODE REVIEW

File: printer_mods/hartk1213/Voron2.4_Pins_Mod/README.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+To complete this mod the following files will need to be reprinted 
+
+**A and B drive units**
+
+ - a_drive_frame_lower_pinned.stl 
+ - a_drive_frame_upper_pinned.stl 
+ - b_drive_frame_lower_pinned.stl 
+ - b_drive_frame_upper_pinned.stl
+
+**Front Idlers**
```
Comment (KiloQubit) [inline printer_mods/hartk1213/Voron2.4_Pins_Mod/README.md:17]: This directory also includes `{a,b}_idler_{lower,upper}.stl`, are those 4 STLs supposed to be printed too?  Or are they the same as stock?  If same as stock, then remove them from the mod.

File: printer_mods/hartk1213/Voron2.4_Pins_Mod/README.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+- [a]_a_tensioner_pinned.stl
+- [a]_b_tensioner_pinned.stl
+
+**XY Joints**
+- xy_joint_left_upper_pinned.stl
+- xy_joint_left_lower_pinned.stl
+- xy_joint_right_upper_pinned.stl
+- xy_joint_right_lower_pinned.stl
+
+**Z Idlers**
```
Comment (KiloQubit) [inline printer_mods/hartk1213/Voron2.4_Pins_Mod/README.md:27]: This directory also includes `z_tensioner_bracker_{a,b}_x2.stl`, are those 2 STLs supposed to be printed too?   Or are they the same as stock?  If same as stock, then remove them from the mod.

**Review by KiloQubit — CHANGES_REQUESTED** [review]
Thanks for your PR!

Please remove the space from `Z Idler_Pinned.step` filename.

I also have some questions below about parts that are possibly still stock - if they're stock, they should be removed from the mod.

File: printer_mods/hartk1213/Voron2.4_Pins_Mod/README.md (hunk 2 above)
Comment (hartk1213) [reply printer_mods/hartk1213/Voron2.4_Pins_Mod/README.md:27]: these have also been removed

File: printer_mods/hartk1213/Voron2.4_Pins_Mod/README.md (hunk 1 above)
Comment (hartk1213) [reply printer_mods/hartk1213/Voron2.4_Pins_Mod/README.md:17]: shoot you're right those are stock my bad, they have been removed

**Comment by KiloQubit** [comment]
There's still a space in `Z Idler_Pinned.step`, please rename the file to not have spaces.

**Review by KiloQubit — APPROVED** [review]
Thanks for the updates!

**Pull request merged**